crates/task-impls/src/vid.rs

```diff
@@ -105,26 +105,19 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
 
             HotShotEvent::BlockReady(vid_disperse, view_number) => {
                 let view_number = *view_number;
-                let Ok(signature) = TYPES::SignatureKey::sign(
-                    &self.private_key,
-                    vid_disperse.payload_commitment.as_ref().as_ref(),
-                ) else {
-                    error!("VID: failed to sign dispersal payload");
-                    return None;
-                };
-                debug!("publishing VID disperse for view {}", *view_number);
-                broadcast_event(
-                    Arc::new(HotShotEvent::VidDisperseSend(
-                        Proposal {
-                            signature,
-                            data: vid_disperse.clone(),
-                            _pd: PhantomData,
-                        },
-                        self.public_key.clone(),
-                    )),
-                    &event_stream,
-                )
-                .await;
+                let mut broadcast_tasks = Vec::new();
+                let vid_disperse_shares = VidDisperseShare::from_vid_disperse(vid_disperse.clone());
+                for vid_disperse_share in vid_disperse_shares {
+                    broadcast_tasks.push(broadcast_event(
+                        Arc::new(HotShotEvent::VidDisperseSend(
+                            vid_disperse_share.to_proposal(&self.private_key)?,
+                            self.public_key.clone(),
+                        )),
+                        &event_stream,
+                    ));
+                }
+                debug!("publishing VID disperse shares for view {}", *view_number);
+                join_all(broadcast_tasks).await;
```

Comment:
This is just creating a bunch of tasks which write to the broadcast queue,  It maybe makes more sense to just send all the shares + signature in the event and handle the splitting and sending on the network side.